wgsl/index.bs

```diff
@@ -9286,6 +9286,247 @@ Any node which is not visited by these reachability analyses can be proven to be
 
 Note: The bottom-up analysis is still required, as it lets us know what edges to add to the graphs when encountering calls.
 
+### Examples ### {#uniformity-examples}
+
+The graphs in the subsequent example use the following conventions for nodes:
+* Rectangles represent value nodes.
+* Rounded  rectangles represent control flow nodes.
+
+#### Invalid `textureSample` Function Call #### {#uniformity-example1}
+
+This example shows an invalid use of a [[#texturesample|textureSample]]
+built-in function call.
+The function call is made within an if statement whose condition depends on a
+non-uniform value (i.e. the built-in value `position`).
+The invalid dependency chain is highlighted in red.
+
+<div class='example wgsl' heading='WGSL invalid textureSample'>
+  <xmp highlight='rust'>
+    @group(0) @binding(0) var t : texture_2d<f32>;
+    @group(0) @binding(1) var s : sampler;
+
+    @fragment
+    fn main(@builtin(position) pos : vec4<f32>) {
+      if (pos.x < 0.5) {
+        // Invalid textureSample function call.
+        _ = textureSample(t, s, pos.xy);
+      }
+    }
+  </xmp>
+</div>
+
+<div class=example1>
+  <figure>
+    <figcaption>Uniformity graph</figcaption>
+    <object type="image/svg+xml" data="img/uniformity_1.mmd.svg"></object>
+  </figure>
+</div>
+
+The example also shows that uniformity of the control flow after the if
+statement is the same as the uniformity prior to the if statement (CF_return
+being connected to CF_start).
+That is, the control flow is once again uniform after the if statement (because
+it is guaranteed to start as uniform control flow at the beginning of the entry
+point).
+If the `textureSample` function call been moved outside the if statement the
+program would have been valid.
+Likewise, if the condition of the if statement were a uniform value (e.g. each
+invocation read the same value from a [=uniform buffer=]), the program would
+also have been valid.
+
+#### Function-scope Variable Uniformity #### {#uniformity-example2}
+
+This example show both a valid and an invalid
+[[#sync-builtin-functions|barrier]] function call that depend on the value of a
+function-scope variable.
+The `workgroupBarrier` is invalid because the value of `x` is derived from the
+mutable module-scope variable `a`.
+The `storageBarrier` is valid because the value of `x` is derived from the
+immutable module-scope variable `b`. 
+This example highlights the [[#func-var-value-analysis|value analysis']]
+ability to separate different periods of uniformity in a function-scope
+variable's lifetime.
+This example also clearly shows that uniformity returns to the control flow
```

Comment:
Suggest rewording to

> shows that control flow becomes uniform again after the the end of the  first if statement.  We know this because that section of the graph...